pandas/core/indexing.py

```diff
@@ -1840,6 +1840,17 @@ def _setitem_single_column(self, loc: int, value, plane_indexer):
         """
         pi = plane_indexer
 
+        if not hasattr(self.obj._mgr, "blocks"):
+            # ArrayManager: in this case we cannot rely on getting the column
+            # as a Series to mutate, but need to operated on the mgr directly
+            if com.is_null_slice(pi) or com.is_full_slice(pi, len(self.obj)):
```

Comment:
xref #44353 (doesn't need to be resolved for this PR, but will make lots of things easier)